app/models/loan.rb

```diff
@@ -87,9 +87,9 @@ def self.filter_by_params(params)
     scoped = scoped.status(params[:status]) if params[:status] != 'all'
 
     if params[:division] != 'all'
-      division_ids = Division.find_by(short_name: params[:division])&.self_and_descendant_ids
+      division_ids = Division.find_by(short_name: params[:division])&.self_and_descendants&.select(&:public?)&.map(&:id)
     else
-      division_ids = Division.root.descendant_ids
+      division_ids = Division.root.descendants.select(&:public?)&.map(&:id)
```

Comment:
I _think_ this would be cleaner and a little more efficient, but should still work. In this version you are only using DB queries, so there's no iterating over ruby methods. In theory this works, because `closure_tree` returns chainable scopes.

```ruby
if params[:division] != 'all'
  division_ids = Division.find_by(short_name: params[:division])&.self_and_descendants&.where(public: true)&.pluck(:id)
else
  division_ids = Division.root.descendants.&where(public: true).&pluck(:id)
end
```